<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Philipp Bayer" />

<meta name="date" content="2020-09-18" />

<title>R-gene count comparisons</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">R gene analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/philippbayer/R_gene_analysis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">R-gene count comparisons</h1>
<h4 class="author">Philipp Bayer</h4>
<h4 class="date">2020-09-18</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-03-30
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>R_gene_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2.9000). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200917code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200917)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200917code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200917)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomphilippbayerRgeneanalysistreee1faf814db55bfd50586477fa1912cec4e75ac65targetblanke1faf81a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/philippbayer/R_gene_analysis/tree/e1faf814db55bfd50586477fa1912cec4e75ac65" target="_blank">e1faf81</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomphilippbayerRgeneanalysistreee1faf814db55bfd50586477fa1912cec4e75ac65targetblanke1faf81a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/philippbayer/R_gene_analysis/tree/e1faf814db55bfd50586477fa1912cec4e75ac65" target="_blank">e1faf81</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  data/Brec_R1.txt
    Untracked:  data/Brec_R2.txt
    Untracked:  data/CR15_R1.txt
    Untracked:  data/CR15_R2.txt
    Untracked:  data/CR_14_R1.txt
    Untracked:  data/CR_14_R2.txt
    Untracked:  data/KS_R1.txt
    Untracked:  data/KS_R2.txt
    Untracked:  data/NBS_PAV.txt.gz
    Untracked:  data/NLR_PAV_GD.txt
    Untracked:  data/NLR_PAV_GM.txt
    Untracked:  data/PAVs_newick.txt
    Untracked:  data/PPR1.txt
    Untracked:  data/PPR2.txt
    Untracked:  data/SNPs_newick.txt
    Untracked:  data/bac.txt
    Untracked:  data/brown.txt
    Untracked:  data/cy3.txt
    Untracked:  data/cy5.txt
    Untracked:  data/early.txt
    Untracked:  data/flowerings.txt
    Untracked:  data/foregeye.txt
    Untracked:  data/height.txt
    Untracked:  data/late.txt
    Untracked:  data/mature.txt
    Untracked:  data/motting.txt
    Untracked:  data/mvp.kin.bin
    Untracked:  data/mvp.kin.desc
    Untracked:  data/oil.txt
    Untracked:  data/pdh.txt
    Untracked:  data/protein.txt
    Untracked:  data/rust_tan.txt
    Untracked:  data/salt.txt
    Untracked:  data/seedq.txt
    Untracked:  data/seedweight.txt
    Untracked:  data/stem_termination.txt
    Untracked:  data/sudden.txt
    Untracked:  data/virus.txt
    Untracked:  output/GAPIT/GAPIT.FarmCPU.Pred.result.csv
    Untracked:  output/GAPIT/GAPIT.FarmCPU.Yield.Df.tValue.StdErr.csv
    Untracked:  output/GAPIT/GAPIT.FarmCPU.Yield.GWAS.Results.csv
    Untracked:  output/GAPIT/GAPIT.FarmCPU.Yield.MAF.pdf
    Untracked:  output/GAPIT/GAPIT.FarmCPU.Yield.Manhattan.Plot.Chromosomewise.pdf
    Untracked:  output/GAPIT/GAPIT.FarmCPU.Yield.Manhattan.Plot.Genomewise.pdf
    Untracked:  output/GAPIT/GAPIT.FarmCPU.Yield.QQ-Plot.pdf
    Untracked:  output/GAPIT/GAPIT.FarmCPU.Yield.phenotype_view.pdf
    Untracked:  output/GAPIT/GAPIT.GLM.Yield.Df.tValue.StdErr.csv
    Untracked:  output/GAPIT/GAPIT.GLM.Yield.GWAS.Results.csv
    Untracked:  output/GAPIT/GAPIT.GLM.Yield.Log.csv
    Untracked:  output/GAPIT/GAPIT.GLM.Yield.MAF.pdf
    Untracked:  output/GAPIT/GAPIT.GLM.Yield.Manhattan.Plot.Chromosomewise.pdf
    Untracked:  output/GAPIT/GAPIT.GLM.Yield.Manhattan.Plot.Genomewise.pdf
    Untracked:  output/GAPIT/GAPIT.GLM.Yield.Optimum.pdf
    Untracked:  output/GAPIT/GAPIT.GLM.Yield.PRED.csv
    Untracked:  output/GAPIT/GAPIT.GLM.Yield.QQ-Plot.pdf
    Untracked:  output/GAPIT/GAPIT.GLM.Yield.ROC.csv
    Untracked:  output/GAPIT/GAPIT.GLM.Yield.ROC.pdf
    Untracked:  output/GAPIT/GAPIT.GLM.Yield.phenotype_view.pdf
    Untracked:  output/GAPIT/GAPIT.MLM.Yield.Df.tValue.StdErr.csv
    Untracked:  output/GAPIT/GAPIT.MLM.Yield.GWAS.Results.csv
    Untracked:  output/GAPIT/GAPIT.MLM.Yield.Log.csv
    Untracked:  output/GAPIT/GAPIT.MLM.Yield.MAF.pdf
    Untracked:  output/GAPIT/GAPIT.MLM.Yield.Manhattan.Plot.Chromosomewise.pdf
    Untracked:  output/GAPIT/GAPIT.MLM.Yield.Manhattan.Plot.Genomewise.pdf
    Untracked:  output/GAPIT/GAPIT.MLM.Yield.Optimum.pdf
    Untracked:  output/GAPIT/GAPIT.MLM.Yield.PRED.csv
    Untracked:  output/GAPIT/GAPIT.MLM.Yield.QQ-Plot.pdf
    Untracked:  output/GAPIT/GAPIT.MLM.Yield.ROC.csv
    Untracked:  output/GAPIT/GAPIT.MLM.Yield.ROC.pdf
    Untracked:  output/GAPIT/GAPIT.MLM.Yield.phenotype_view.pdf
    Untracked:  output/GAPIT/GAPIT.MLMM.Pred.result.csv
    Untracked:  output/GAPIT/GAPIT.MLMM.Yield.Df.tValue.StdErr.csv
    Untracked:  output/GAPIT/GAPIT.MLMM.Yield.GWAS.Results.csv
    Untracked:  output/GAPIT/GAPIT.MLMM.Yield.MAF.pdf
    Untracked:  output/GAPIT/GAPIT.MLMM.Yield.Manhattan.Plot.Chromosomewise.pdf
    Untracked:  output/GAPIT/GAPIT.MLMM.Yield.Manhattan.Plot.Genomewise.pdf
    Untracked:  output/GAPIT/GAPIT.MLMM.Yield.QQ-Plot.pdf
    Untracked:  output/GAPIT/GAPIT.MLMM.Yield.phenotype_view.pdf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/gwas.Rmd</code>) and HTML (<code>docs/gwas.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/philippbayer/R_gene_analysis/blob/e1faf814db55bfd50586477fa1912cec4e75ac65/analysis/gwas.Rmd" target="_blank">e1faf81</a>
</td>
<td>
Philipp Bayer
</td>
<td>
2021-03-30
</td>
<td>
wflow_publish(c(“analysis/gwas.Rmd”, “analysis/index.Rmd”))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/philippbayer/R_gene_analysis/c7696734f40c6d9a9e8af99a5dc18265232e6a9a/docs/gwas.html" target="_blank">c769673</a>
</td>
<td>
Philipp Bayer
</td>
<td>
2021-03-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/philippbayer/R_gene_analysis/blob/764ad7be589668fa4e0ab14de15e4ab07b2c91a9/analysis/gwas.Rmd" target="_blank">764ad7b</a>
</td>
<td>
Philipp Bayer
</td>
<td>
2021-03-29
</td>
<td>
wflow_publish(“analysis/gwas.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/philippbayer/R_gene_analysis/blob/bd5506f26293d7ad95181aa775685cc6288deadb/analysis/gwas.Rmd" target="_blank">bd5506f</a>
</td>
<td>
Philipp Bayer
</td>
<td>
2021-03-29
</td>
<td>
missing commits
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/philippbayer/R_gene_analysis/2844fd34451df722463e34e8be8d95a79e0ef747/docs/gwas.html" target="_blank">2844fd3</a>
</td>
<td>
Philipp Bayer
</td>
<td>
2021-02-09
</td>
<td>
Add changes, add missing yield
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/philippbayer/R_gene_analysis/e5191225ceaf2f4c7e8136b5af7d4385964814cc/docs/gwas.html" target="_blank">e519122</a>
</td>
<td>
Philipp Bayer
</td>
<td>
2020-12-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/philippbayer/R_gene_analysis/83dc49a3c8bdb0399ad0004ee65fedea770055b2/docs/gwas.html" target="_blank">83dc49a</a>
</td>
<td>
Philipp Bayer
</td>
<td>
2020-12-11
</td>
<td>
Trace deleted files
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/philippbayer/R_gene_analysis/blob/dae157bbfbf4210c5a5e22ecdddffb0eebff9a35/analysis/gwas.Rmd" target="_blank">dae157b</a>
</td>
<td>
Philipp Bayer
</td>
<td>
2020-09-24
</td>
<td>
Update of analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/philippbayer/R_gene_analysis/dae157bbfbf4210c5a5e22ecdddffb0eebff9a35/docs/gwas.html" target="_blank">dae157b</a>
</td>
<td>
Philipp Bayer
</td>
<td>
2020-09-24
</td>
<td>
Update of analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Let’s run a GWAS via GAPIT using the NLR genes as input, and the PCs calculated using SNPs</p>
<pre class="r"><code>library(GAPIT3)
library(SNPRelate)
library(tidyverse)
library(ggsignif)
library(cowplot)
library(patchwork)</code></pre>
<p>First, we have to make the principal components - making them based on NLR genes alone is probably garbage. Let’s make them based on all publicly available SNPs from <a href="https://research-repository.uwa.edu.au/en/datasets/soybean-pangenome-snps-assembly-annotation-pav">here</a>.</p>
<p>I ran the following on a bigger server which is why it’s marked as not to run when I rerun workflowr.</p>
<pre class="r"><code>if (!file.exists(&#39;data/SNPs_lee.id.biallic_maf_0.05_geno_0.1.vcf.gz&#39;)) {
  download.file(&#39;https://research-repository.uwa.edu.au/files/89232545/SNPs_lee.id.biallic_maf_0.05_geno_0.1.vcf.gz&#39;, &#39;data/SNPs_lee.id.biallic_maf_0.05_geno_0.1.vcf.gz&#39;)
}

#We have to convert the big vcf file

if (!file.exists(&quot;data/snp.gds&quot;)) {
  vcf.fn &lt;- &#39;data/SNPs_lee.id.biallic_maf_0.05_geno_0.1.vcf.gz&#39;
  snpgdsVCF2GDS(vcf.fn, &quot;data/snp.gds&quot;, method=&quot;biallelic.only&quot;)
}

genofile &lt;- snpgdsOpen(&#39;data/snp.gds&#39;)

# Let&#39;s prune the SNPs based on 0.2 LD

snpset &lt;- snpgdsLDpruning(genofile, ld.threshold=0.2, autosome.only=F)
snpset.id &lt;- unlist(unname(snpset))

# And now let&#39;s run the PCA
pca &lt;- snpgdsPCA(genofile, snp.id=snpset.id, num.thread=2, autosome.only=F)

saveRDS(pca, &#39;data/pca.rds&#39;)</code></pre>
<p>OK let’s load the results I made on the remote server and saved in a file:</p>
<pre class="r"><code># load PCA
pca &lt;- readRDS(&#39;data/pca.rds&#39;)</code></pre>
<p>A quick diagnostic plot</p>
<pre class="r"><code>tab &lt;- data.frame(sample.id = pca$sample.id,
    EV1 = pca$eigenvect[,1],    # the first eigenvector
    EV2 = pca$eigenvect[,2],    # the second eigenvector
    stringsAsFactors = FALSE)
plot(tab$EV2, tab$EV1, xlab=&quot;eigenvector 2&quot;, ylab=&quot;eigenvector 1&quot;)</code></pre>
<p><img src="figure/gwas.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>head(tab)</code></pre>
<pre><code>  sample.id         EV1          EV2
1     AB-01 -0.03587935  0.008552839
2     AB-02 -0.03681818  0.010897831
3     BR-01 -0.01975551 -0.016854610
4     BR-02 -0.01651350 -0.015271782
5     BR-03 -0.01702049 -0.018778840
6     BR-04 -0.01639033 -0.021407320</code></pre>
<p>Alright, now we have SNP-based principal components. We’ll use the first two as our own covariates in GAPIT.</p>
<p>I wrote a Python script which takes the NLR-only PAV table and turns that into HapMap format with fake SNPs, see <code>code/transformToGAPIT.py</code>.</p>
<pre class="r"><code>myY &lt;- read.table(&#39;data/yield.txt&#39;, head = TRUE)
myGD &lt;- read.table(&#39;data/NLR_PAV_GD.txt&#39;, head = TRUE)
myGM &lt;- read.table(&#39;data/NLR_PAV_GM.txt&#39;, head = TRUE)</code></pre>
<p>GAPIT prints a LOT of stuff so I turn that off here, what’s important are all the output files.</p>
<pre class="r"><code>myGAPIT &lt;- GAPIT(
  Y=myY[,c(1,2)],
  CV = tab,
  GD = myGD,
  GM = myGM,
  PCA.total = 0, # turn off PCA calculation as I use my own based on SNPs
  model = c(&#39;GLM&#39;, &#39;MLM&#39;, &#39;MLMM&#39;, &#39;FarmCPU&#39;)
)</code></pre>
<pre class="r"><code>if (! dir.exists(&#39;output/GAPIT&#39;)){
  dir.create(&#39;output/GAPIT&#39;)
}</code></pre>
<p>R doesn’t have an in-built function to move files, so I copy and delete the output files here. There’s a package which adds file-moving but I’m not adding a whole dependency just for one convenience function, I’m not a Node.js person ;)</p>
<pre class="r"><code>for(file in list.files(&#39;.&#39;, pattern=&#39;GAPIT*&#39;)) {
  file.copy(file, &#39;output/GAPIT&#39;)
  file.remove(file)
}</code></pre>
<p>Let’s make a table of the statistically significantly associated SNPs.</p>
<pre class="r"><code>results_files &lt;- list.files(&#39;output/GAPIT/&#39;, pattern=&#39;*GWAS.Results.csv&#39;, full.names = T)</code></pre>
<p>Let’s use FDR &lt; 0.05 as cutoff</p>
<pre class="r"><code>results_df &lt;- NULL
for(i in seq_along(results_files)) {
  this_df &lt;- read_csv(results_files[i])
  filt_df &lt;- this_df %&gt;% filter(`FDR_Adjusted_P-values` &lt; 0.05)
  
  # pull the method name out and add to dataframe
  this_method &lt;- str_split(results_files[i], &quot;\\.&quot;)[[1]][2]
  filt_df &lt;- filt_df %&gt;% add_column(Method = this_method, .before = &#39;SNP&#39;)
  if (is.null(results_df)) {
    results_df &lt;- filt_df
  } else {
    results_df &lt;- rbind(results_df, filt_df)
  }
}</code></pre>
<pre><code>Parsed with column specification:
cols(
  SNP = col_character(),
  Chromosome = col_double(),
  Position = col_double(),
  P.value = col_double(),
  maf = col_double(),
  nobs = col_double(),
  Rsquare.of.Model.without.SNP = col_logical(),
  Rsquare.of.Model.with.SNP = col_logical(),
  `FDR_Adjusted_P-values` = col_double(),
  effect = col_double()
)</code></pre>
<pre><code>Parsed with column specification:
cols(
  SNP = col_character(),
  Chromosome = col_double(),
  Position = col_double(),
  P.value = col_double(),
  maf = col_double(),
  nobs = col_double(),
  Rsquare.of.Model.without.SNP = col_double(),
  Rsquare.of.Model.with.SNP = col_double(),
  `FDR_Adjusted_P-values` = col_double(),
  effect = col_double()
)
Parsed with column specification:
cols(
  SNP = col_character(),
  Chromosome = col_double(),
  Position = col_double(),
  P.value = col_double(),
  maf = col_double(),
  nobs = col_double(),
  Rsquare.of.Model.without.SNP = col_double(),
  Rsquare.of.Model.with.SNP = col_double(),
  `FDR_Adjusted_P-values` = col_double(),
  effect = col_double()
)</code></pre>
<pre><code>Parsed with column specification:
cols(
  SNP = col_character(),
  Chromosome = col_double(),
  Position = col_double(),
  P.value = col_double(),
  maf = col_double(),
  nobs = col_double(),
  Rsquare.of.Model.without.SNP = col_logical(),
  Rsquare.of.Model.with.SNP = col_logical(),
  `FDR_Adjusted_P-values` = col_double(),
  effect = col_logical()
)</code></pre>
<pre class="r"><code>results_df</code></pre>
<pre><code># A tibble: 15 x 11
   Method SNP   Chromosome Position P.value     maf  nobs Rsquare.of.Mode~
   &lt;chr&gt;  &lt;chr&gt;      &lt;dbl&gt;    &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt; &lt;dbl&gt;            &lt;dbl&gt;
 1 FarmC~ Glym~          2 49198872 3.85e-6 0.00130   769           NA    
 2 FarmC~ UWAS~         22      277 3.73e-5 0.104     769           NA    
 3 FarmC~ UWAS~         22      405 8.13e-5 0.124     769           NA    
 4 FarmC~ Glym~          1  3567950 1.90e-4 0.493     769           NA    
 5 FarmC~ UWAS~         22      326 2.72e-4 0.198     769           NA    
 6 FarmC~ Glym~         14  1789365 4.22e-4 0.224     769           NA    
 7 GLM    Glym~          2 49198872 1.19e-8 0.00130   769            0.380
 8 GLM    UWAS~         22      277 1.50e-5 0.104     769            0.380
 9 GLM    UWAS~         22      326 3.15e-5 0.198     769            0.380
10 GLM    UWAS~         22      405 4.46e-5 0.124     769            0.380
11 GLM    Glym~         16 30996392 1.21e-4 0.0676    769            0.380
12 GLM    UWAS~         22      958 3.78e-4 0.0364    769            0.380
13 GLM    Glym~          1  3567950 3.85e-4 0.493     769            0.380
14 MLM    Glym~          2 49198872 5.32e-7 0.00130   769            0.422
15 MLMM   Glym~          2 49198872 2.83e-7 0.00130   769           NA    
# ... with 3 more variables: Rsquare.of.Model.with.SNP &lt;dbl&gt;,
#   `FDR_Adjusted_P-values` &lt;dbl&gt;, effect &lt;dbl&gt;</code></pre>
<p>Ah beautiful, one NLR gene found by all methods, and a bunch of extra pangenome genes found by FarmCPU/GLM. However the one gene found by all methods has a horrible MAF so we’ll probably end up ignoring that.</p>
<pre class="r"><code>results_df %&gt;% group_by(SNP) %&gt;% count() %&gt;% arrange(n)</code></pre>
<pre><code># A tibble: 8 x 2
# Groups:   SNP [8]
  SNP                        n
  &lt;chr&gt;                  &lt;int&gt;
1 GlymaLee.14G022000.1.p     1
2 GlymaLee.16G111300.1.p     1
3 UWASoyPan00022             1
4 GlymaLee.01G030900.1.p     2
5 UWASoyPan00316             2
6 UWASoyPan00772             2
7 UWASoyPan04354             2
8 GlymaLee.02G228600.1.p     4</code></pre>
<pre class="r"><code>candidates &lt;- results_df %&gt;% group_by(SNP) %&gt;% count() %&gt;% filter(n &gt;= 2)
candidates</code></pre>
<pre><code># A tibble: 5 x 2
# Groups:   SNP [5]
  SNP                        n
  &lt;chr&gt;                  &lt;int&gt;
1 GlymaLee.01G030900.1.p     2
2 GlymaLee.02G228600.1.p     4
3 UWASoyPan00316             2
4 UWASoyPan00772             2
5 UWASoyPan04354             2</code></pre>
<p>Let’s focus on genes found by more than one method.</p>
<p>This particular R-gene GlymaLee.02G228600.1.p seems to have a strong negative impact on yield (effect -1.69 in MLM, -1.94 in GLM, -1.72 in FarmCPU) but it also has a horrible MAF, normally this would get filtered in SNPs but we don’t have SNPs here. Let’s plot the yield for individuals who have vs those who don’t have those genes that appear in more than one method.</p>
<pre class="r"><code>pav_table &lt;- read_tsv(&#39;./data/soybean_pan_pav.matrix_gene.txt.gz&#39;)</code></pre>
<pre><code>Parsed with column specification:
cols(
  .default = col_double(),
  Individual = col_character()
)</code></pre>
<pre><code>See spec(...) for full column specifications.</code></pre>
<pre class="r"><code>yield &lt;- read_tsv(&#39;./data/yield.txt&#39;)</code></pre>
<pre><code>Parsed with column specification:
cols(
  Line = col_character(),
  Yield = col_double()
)</code></pre>
<pre class="r"><code>pav_table %&gt;% 
  filter(Individual == &#39;GlymaLee.02G228600.1.p&#39;) %&gt;% 
  pivot_longer(!Individual, names_to = &#39;Line&#39;, values_to = &#39;Presence&#39;) %&gt;% 
  inner_join(yield) %&gt;% arrange(Presence) %&gt;% head()</code></pre>
<pre><code>Joining, by = &quot;Line&quot;</code></pre>
<pre><code># A tibble: 6 x 4
  Individual             Line    Presence Yield
  &lt;chr&gt;                  &lt;chr&gt;      &lt;dbl&gt; &lt;dbl&gt;
1 GlymaLee.02G228600.1.p USB-762        0  2.88
2 GlymaLee.02G228600.1.p AB-01          1  1.54
3 GlymaLee.02G228600.1.p AB-02          1  1.3 
4 GlymaLee.02G228600.1.p For            1  3.34
5 GlymaLee.02G228600.1.p HN001          1  3.54
6 GlymaLee.02G228600.1.p HN005          1  2.15</code></pre>
<p>OK this gene is ‘boring’ - it’s lost only in a single line, and we can’t trust that. We’ll remove it.</p>
<pre class="r"><code>pav_table %&gt;% 
  filter(Individual == &#39;GlymaLee.02G228600.1.p&#39;) %&gt;% 
  pivot_longer(!Individual, names_to = &#39;Line&#39;, values_to = &#39;Presence&#39;) %&gt;% 
  inner_join(yield) %&gt;% summarise(median_y= median(Yield))</code></pre>
<pre><code>Joining, by = &quot;Line&quot;</code></pre>
<pre><code># A tibble: 1 x 1
  median_y
     &lt;dbl&gt;
1     2.18</code></pre>
<p>At least this particular line with the lost gene has slightly higher yield than the median? but soooo many different reasons… Let’s remove this gene as MAF &lt; 5%.</p>
<p>Let’s check the other genes.</p>
<pre class="r"><code>candidates &lt;- candidates %&gt;% filter(SNP != &#39;GlymaLee.02G228600.1.p&#39;)</code></pre>
<pre class="r"><code>plots &lt;- list()
for(i in 1:nrow(candidates)) {
  this_cand = candidates[i,]
  p &lt;- pav_table %&gt;% 
    filter(Individual == this_cand$SNP) %&gt;% 
    pivot_longer(!Individual, names_to = &#39;Line&#39;, values_to = &#39;Presence&#39;) %&gt;% 
    inner_join(yield) %&gt;% 
    mutate(Presence = case_when(
      Presence == 0.0 ~ &#39;Lost&#39;,
      Presence == 1.0 ~ &#39;Present&#39;
    )) %&gt;% 
    ggplot(aes(x=Presence, y = Yield, group=Presence)) + 
    geom_boxplot() + 
    geom_jitter(alpha=0.9, size=0.4) + 
    geom_signif(comparisons = list(c(&#39;Lost&#39;, &#39;Present&#39;)), 
              map_signif_level = T) +
    theme_minimal_hgrid()
    #xlab(paste(&#39;Presence&#39;, str_replace(this_cand$SNP, &#39;.1.p&#39;,&#39;&#39;))) # make the gene name a bit nicer
  plots[[i]] &lt;- p
}</code></pre>
<pre><code>Joining, by = &quot;Line&quot;
Joining, by = &quot;Line&quot;
Joining, by = &quot;Line&quot;
Joining, by = &quot;Line&quot;</code></pre>
<pre class="r"><code>wrap_plots(plots) + plot_annotation(tag_levels = &#39;A&#39;)</code></pre>
<p><img src="figure/gwas.Rmd/unnamed-chunk-17-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>Four nice plots we have now, we’ll use those as supplementary. It’s interesting how the reference gene has a positive impact on yield, the pangenome extra gene has a negative impact, and the other two don’t do much. Then again there are many reasons why we see this outcome.</p>
<p>For myself, the four labels: GlymaLee.01G030900.1.p, UWASoyPan00316, UWASoyPan00772, UWASoyPan04354</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.3 (2020-02-29)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 17134)

Matrix products: default

locale:
[1] LC_COLLATE=English_Australia.1252  LC_CTYPE=English_Australia.1252   
[3] LC_MONETARY=English_Australia.1252 LC_NUMERIC=C                      
[5] LC_TIME=English_Australia.1252    

attached base packages:
[1] compiler  parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] biganalytics_1.1.21  biglm_0.9-2          DBI_1.1.0           
 [4] foreach_1.4.8        bigmemory_4.5.36     scatterplot3d_0.3-41
 [7] EMMREML_3.1          Matrix_1.2-18        ape_5.3             
[10] genetics_1.3.8.1.2   mvtnorm_1.1-1        MASS_7.3-51.6       
[13] gtools_3.8.1         gdata_2.18.0         combinat_0.0-8      
[16] LDheatmap_0.99-8     gplots_3.0.3         multtest_2.42.0     
[19] Biobase_2.46.0       BiocGenerics_0.32.0  patchwork_1.0.0     
[22] cowplot_1.0.0        ggsignif_0.6.0       forcats_0.5.0       
[25] stringr_1.4.0        dplyr_1.0.0          purrr_0.3.4         
[28] readr_1.3.1          tidyr_1.1.0          tibble_3.0.2        
[31] ggplot2_3.3.2        tidyverse_1.3.0      SNPRelate_1.20.1    
[34] gdsfmt_1.22.0        GAPIT3_3.1.0         workflowr_1.6.2.9000

loaded via a namespace (and not attached):
 [1] nlme_3.1-148        bitops_1.0-6        fs_1.5.0.9000      
 [4] lubridate_1.7.9     httr_1.4.2          rprojroot_1.3-2    
 [7] tools_3.6.3         backports_1.1.10    utf8_1.1.4         
[10] R6_2.4.1            KernSmooth_2.23-17  colorspace_1.4-1   
[13] withr_2.2.0         tidyselect_1.1.0    processx_3.4.4     
[16] git2r_0.27.1        cli_2.0.2           rvest_0.3.5        
[19] xml2_1.3.2          labeling_0.3        caTools_1.18.0     
[22] scales_1.1.1        callr_3.4.4         digest_0.6.25      
[25] rmarkdown_2.3       pkgconfig_2.0.3     htmltools_0.5.0    
[28] dbplyr_1.4.4        rlang_0.4.7         readxl_1.3.1       
[31] rstudioapi_0.11     farver_2.0.3        generics_0.0.2     
[34] jsonlite_1.7.1      magrittr_1.5        Rcpp_1.0.5         
[37] munsell_0.5.0       fansi_0.4.1         lifecycle_0.2.0    
[40] stringi_1.5.3       whisker_0.4         yaml_2.2.1         
[43] grid_3.6.3          blob_1.2.1          promises_1.1.1     
[46] bigmemory.sri_0.1.3 crayon_1.3.4        lattice_0.20-41    
[49] haven_2.3.1         splines_3.6.3       hms_0.5.3          
[52] knitr_1.29          ps_1.3.4            pillar_1.4.4       
[55] codetools_0.2-16    stats4_3.6.3        reprex_0.3.0       
[58] glue_1.4.2          evaluate_0.14       getPass_0.2-2      
[61] modelr_0.1.8        vctrs_0.3.1         httpuv_1.5.4       
[64] cellranger_1.1.0    gtable_0.3.0        assertthat_0.2.1   
[67] xfun_0.17           broom_0.5.6         later_1.1.0.1      
[70] survival_3.2-3      iterators_1.0.12    ellipsis_0.3.1     </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
