---
title: "R-gene count comparisons"
author: "Philipp Bayer"
date: "2020-09-18"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

Let's run a GWAS via GAPIT using the NLR genes as input, and the PCs calculated using SNPs

```{r setup, warning=FALSE, message=FALSE}
library(GAPIT3)
library(SNPRelate)
```

First, we have to make the principal components - making them based on NLR genes alone is probably garbage. Let's make them based on all publicly available SNPs from [here](https://research-repository.uwa.edu.au/en/datasets/soybean-pangenome-snps-assembly-annotation-pav).

I ran the following on a bigger server which is why it's marked as not to run when I rerun workflowr.

```{r eval=FALSE}
if (!file.exists('data/SNPs_lee.id.biallic_maf_0.05_geno_0.1.vcf.gz')) {
  download.file('https://research-repository.uwa.edu.au/files/89232545/SNPs_lee.id.biallic_maf_0.05_geno_0.1.vcf.gz', 'data/SNPs_lee.id.biallic_maf_0.05_geno_0.1.vcf.gz')
}

#We have to convert the big vcf file

if (!file.exists("data/snp.gds")) {
  vcf.fn <- 'data/SNPs_lee.id.biallic_maf_0.05_geno_0.1.vcf.gz'
  snpgdsVCF2GDS(vcf.fn, "data/snp.gds", method="biallelic.only")
}

genofile <- snpgdsOpen('data/snp.gds')

# Let's prune the SNPs based on 0.2 LD

snpset <- snpgdsLDpruning(genofile, ld.threshold=0.2)
snpset.id <- unlist(unname(snpset))

# And now let's run the PCA
pca <- snpgdsPCA(genofile, snp.id=snpset.id, num.thread=2)
```

OK let's load the results I made on the remote server and saved in a file:

```{r}
# load PCA
```

A quick diagnostic plot

```{r}
plot(tab$EV2, tab$EV1, xlab="eigenvector 2", ylab="eigenvector 1")
```

Alright, now we have SNP-based principal components. We'll use the first two as our own covariates in GAPIT.




Now we can delete the big vcf file and the big gds file

```{r}
file.remove(vcf.fn)
file.remove('data/snp.gds')
```
