---
title: "R-gene count comparisons"
author: "Philipp Bayer"
date: "2020-09-18"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

Let's run a GWAS via GAPIT using the NLR genes as input, and the PCs calculated using SNPs

```{r setup, warning=FALSE, message=FALSE}
library(GAPIT3)
library(SNPRelate)
library(ggtext)
library(tidyverse)
library(ggsignif)
library(cowplot)
library(ggsci)
library(patchwork)
```

First, we have to make the principal components - making them based on NLR genes alone is probably garbage. Let's make them based on all publicly available SNPs from [here](https://research-repository.uwa.edu.au/en/datasets/soybean-pangenome-snps-assembly-annotation-pav).

I ran the following on a bigger server which is why it's marked as not to run when I rerun workflowr.

```{r eval=FALSE}
if (!file.exists('data/SNPs_lee.id.biallic_maf_0.05_geno_0.1.vcf.gz')) {
  download.file('https://research-repository.uwa.edu.au/files/89232545/SNPs_lee.id.biallic_maf_0.05_geno_0.1.vcf.gz', 'data/SNPs_lee.id.biallic_maf_0.05_geno_0.1.vcf.gz')
}

#We have to convert the big vcf file

if (!file.exists("data/snp.gds")) {
  vcf.fn <- 'data/SNPs_lee.id.biallic_maf_0.05_geno_0.1.vcf.gz'
  snpgdsVCF2GDS(vcf.fn, "data/snp.gds", method="biallelic.only")
}

genofile <- snpgdsOpen('data/snp.gds')

# Let's prune the SNPs based on 0.2 LD

snpset <- snpgdsLDpruning(genofile, ld.threshold=0.2, autosome.only=F)
snpset.id <- unlist(unname(snpset))

# And now let's run the PCA
pca <- snpgdsPCA(genofile, snp.id=snpset.id, num.thread=2, autosome.only=F)

saveRDS(pca, 'data/pca.rds')
```

OK let's load the results I made on the remote server and saved in a file:

```{r}
# load PCA
pca <- readRDS('data/pca.rds')
```

A quick diagnostic plot

```{r}
tab <- data.frame(sample.id = pca$sample.id,
    EV1 = pca$eigenvect[,1],    # the first eigenvector
    EV2 = pca$eigenvect[,2],    # the second eigenvector
    stringsAsFactors = FALSE)
plot(tab$EV2, tab$EV1, xlab="eigenvector 2", ylab="eigenvector 1")
```

```{r}
head(tab)
```

Alright, now we have SNP-based principal components. We'll use the first two as our own covariates in GAPIT.

I wrote a Python script which takes the NLR-only PAV table and turns that into HapMap format with fake SNPs, see `code/transformToGAPIT.py`.

```{r eval=FALSE}
myY <- read.table('data/yield.txt', head = TRUE)
myGD <- read.table('data/NLR_PAV_GD.txt', head = TRUE)
myGM <- read.table('data/NLR_PAV_GM.txt', head = TRUE)
```

GAPIT prints a LOT of stuff so I turn that off here, what's important are all the output files.

```{r, eval=FALSE, message=FALSE, warning=FALSE, results=FALSE}
myGAPIT <- GAPIT(
  Y=myY[,c(1,2)],
  CV = tab,
  GD = myGD,
  GM = myGM,
  PCA.total = 0, # turn off PCA calculation as I use my own based on SNPs
  model = c('GLM', 'MLM', 'MLMM', 'FarmCPU')
)
```

```{r eval=FALSE}
if (! dir.exists('output/GAPIT')){
  dir.create('output/GAPIT')
}
```

R doesn't have an in-built function to move files, so I copy and delete the output files here. There's a package which adds file-moving but I'm not adding a whole dependency just for one convenience function, I'm not a Node.js person ;) 

```{r eval=FALSE}
for(file in list.files('.', pattern='GAPIT*')) {
  file.copy(file, 'output/GAPIT')
  file.remove(file)
}
```

Let's make a table of the statistically significantly associated SNPs.

```{r}
results_files <- list.files('output/GAPIT/', pattern='*GWAS.Results.csv', full.names = T)
```

Let's use FDR < 0.05 as cutoff

```{r}
results_df <- NULL
for(i in seq_along(results_files)) {
  this_df <- read_csv(results_files[i])
  filt_df <- this_df %>% filter(`FDR_Adjusted_P-values` < 0.05)
  
  # pull the method name out and add to dataframe
  this_method <- str_split(results_files[i], "\\.")[[1]][2]
  filt_df <- filt_df %>% add_column(Method = this_method, .before = 'SNP')
  if (is.null(results_df)) {
    results_df <- filt_df
  } else {
    results_df <- rbind(results_df, filt_df)
  }
}
results_df %>% knitr::kable()
```

Ah beautiful, one NLR gene found by all methods, and a bunch of extra pangenome genes found by FarmCPU/GLM. However the one gene found by all methods has a horrible MAF so we'll probably end up ignoring that.

```{r}
results_df %>% group_by(SNP) %>% count() %>% arrange(n)
candidates <- results_df %>% 
  group_by(SNP) %>% 
  count() %>% 
  filter(n >= 2)

candidates %>% knitr::kable()
```

Let's focus on genes found by more than one method.

This particular R-gene GlymaLee.02G228600.1.p seems to have a strong negative impact on yield (effect -1.69 in MLM, -1.94 in GLM, -1.72 in FarmCPU) but it also has a horrible MAF, normally this would get filtered in SNPs but we don't have SNPs here.  Let's plot the yield for individuals who have vs those who don't have those genes that appear in more than one method.

```{r}
pav_table <- read_tsv('./data/soybean_pan_pav.matrix_gene.txt.gz')
```

```{r}
yield <- read_tsv('./data/yield.txt')
```

```{r}
pav_table %>% 
  filter(Individual == 'GlymaLee.02G228600.1.p') %>% 
  pivot_longer(!Individual, names_to = 'Line', values_to = 'Presence') %>% 
  inner_join(yield) %>% arrange(Presence) %>% head()
```

OK this gene is 'boring' - it's lost only in a single line, and we can't trust that. We'll remove it.

```{r}
pav_table %>% 
  filter(Individual == 'GlymaLee.02G228600.1.p') %>% 
  pivot_longer(!Individual, names_to = 'Line', values_to = 'Presence') %>% 
  inner_join(yield) %>% 
  summarise(median_y= median(Yield))
```

At least this particular line with the lost gene has slightly higher yield than the median? but soooo many different reasons... Let's remove this gene as MAF < 5%.

Let's check the other genes.

```{r}
candidates <- candidates %>% 
  filter(SNP != 'GlymaLee.02G228600.1.p')
candidates %>% knitr::kable()
```

```{r fig.height=8, fig.width=8}
plots <- list()
for(i in 1:nrow(candidates)) {
  this_cand = candidates[i,]
  p <- pav_table %>% 
    filter(Individual == this_cand$SNP) %>% 
    pivot_longer(!Individual, names_to = 'Line', values_to = 'Presence') %>% 
    inner_join(yield) %>% 
    mutate(Presence = case_when(
      Presence == 0.0 ~ 'Lost',
      Presence == 1.0 ~ 'Present'
    )) %>% 
    ggplot(aes(x=Presence, y = Yield, group=Presence)) + 
    geom_boxplot() + 
    geom_jitter(alpha=0.9, size=0.4) + 
    geom_signif(comparisons = list(c('Lost', 'Present')), 
              map_signif_level = T) +
    theme_minimal_hgrid()
    #xlab(paste('Presence', str_replace(this_cand$SNP, '.1.p',''))) # make the gene name a bit nicer
  plots[[i]] <- p
}

wrap_plots(plots) + 
  plot_annotation(tag_levels = 'A')
```

Four nice plots we have now, we'll use those as supplementary. It's interesting how the reference gene has a positive impact on yield, the pangenome extra gene has a negative impact, and the other two don't do much. Then again there are many reasons why we see this outcome.

For myself, the four labels: `r candidates$SNP`

Let's also make those plots, grouped by breeding group
```{r}
groups <- read_csv('./data/Table_of_cultivar_groups.csv')
groups <- groups %>% dplyr::rename(Group = `Group in violin table`)
groups <- groups %>% 
  mutate(Group = str_replace_all(Group, 'landrace', 'Landrace')) %>%
  mutate(Group = str_replace_all(Group, 'Old_cultivar', 'Old cultivar')) %>%
  mutate(Group = str_replace_all(Group, 'Modern_cultivar', 'Modern cultivar')) %>%
  mutate(Group = str_replace_all(Group, 'Wild-type', 'Wild'))

groups$Group <-
  factor(
    groups$Group,
    levels = c('Wild',
               'Landrace',
               'Old cultivar',
               'Modern cultivar')
  )

```

```{r fig.height=10, fig.width=10}
npg_col = pal_npg("nrc")(9)

col_list <- c(`Wild`=npg_col[8],
   Landrace = npg_col[3],
  `Old cultivar`=npg_col[2],
  `Modern cultivar`=npg_col[4])


plots <- list()
for(i in 1:nrow(candidates)) {
  this_cand = candidates[i,]
  p <- pav_table %>% 
    filter(Individual == this_cand$SNP) %>% 
    pivot_longer(!Individual, names_to = 'Line', values_to = 'Presence') %>% 
    inner_join(yield) %>% 
    inner_join(groups, by = c('Line'='Data-storage-ID')) %>% 
    mutate(Presence = case_when(
      Presence == 0.0 ~ 'Lost',
      Presence == 1.0 ~ 'Present'
    )) %>% 
    ggplot(aes(x=Presence, y = Yield, group=Presence, color=Group)) + 
    geom_boxplot() + 
    geom_jitter(alpha=0.9, size=0.4) + 
    facet_wrap(~Group) +
    geom_signif(comparisons = list(c('Lost', 'Present')), 
              map_signif_level = T) +
    theme_minimal_hgrid() +
    ylab(expression(paste('Yield [Mg ', ha ^ -1, ']'))) +
    scale_color_manual(values = col_list)

  plots[[i]] <- p
}

wrap_plots(plots) + 
  plot_annotation(tag_levels = 'A') +
  plot_layout(guides='collect') & theme(legend.position = 'none')
```

I also need a presence/absence percentage for all candidates

```{r}
big_t <- NULL
for(i in 1:nrow(candidates)) {
  this_cand = candidates[i,]
  t <- pav_table %>% 
    filter(Individual == this_cand$SNP) %>% 
    pivot_longer(!Individual, names_to = 'Line', values_to = 'Presence') %>%
    inner_join(groups, by = c('Line'='Data-storage-ID')) %>% group_by(Group) %>% 
    dplyr::select(Presence, Group) %>% 
    table() %>% as_tibble()
  #t <- t %>% add_column(Presence=c(0,1), .before = '.')
  t<- t %>% add_column(Gene=this_cand$SNP, .before='Presence')
  if(is.null(big_t)) {
    big_t <- t
  } else {
    big_t <- rbind(big_t, t)
  }
}

big_t %>% group_by(Group, Gene) %>% mutate(group_sum = sum(n)) %>% 
  mutate(`Percentage` = n/group_sum*100) %>% 
  dplyr::select(-group_sum) %>% 
  rename(Count=n) %>% 
  mutate(Presence = case_when(
      Presence == 0.0 ~ 'Lost',
      Presence == 1.0 ~ 'Present'
    ))  %>%
  knitr::kable(digits=2)

```

```{r}
big_t %>% group_by(Group, Gene) %>% mutate(group_sum = sum(n)) %>%
  mutate(Gene = str_replace_all(Gene, '.1.p','')) %>% 
  mutate(`Percentage` = n/group_sum*100) %>% 
  dplyr::select(-group_sum) %>% 
  filter(Presence == '1') %>% 
  mutate(Group = factor(Group, levels=c('Wild', 'Landrace', 'Old cultivar', 'Modern cultivar'))) %>% 
  ggplot(aes(x = Group, y = Percentage, group=Gene, color=Gene)) + 
  geom_line(size=1.5) +
  theme_minimal_hgrid() +
  ylab('Percentage present') +
  scale_color_brewer(palette = 'Dark2') +
  theme(axis.text.x = element_text(angle = -45, hjust=0))
```

and let's also get the per-gene mean and median yield

```{r}
yields <- NULL
for(i in 1:nrow(candidates)) {
  this_cand = candidates[i,]
  this_t <- pav_table %>% 
    filter(Individual == this_cand$SNP) %>% 
    pivot_longer(!Individual, names_to = 'Line', values_to = 'Presence') %>% 
    inner_join(yield) %>%
    inner_join(groups, by = c('Line'='Data-storage-ID')) %>% 
    group_by(Presence, Group) %>% 
    summarise(`Mean yield` = mean(Yield)) %>% 
    add_column(Gene=this_cand$SNP, .before='Presence') %>% 
    arrange(Group)
  if (is.null(yields)) {
    yields <- this_t
  } else {
    yields <- rbind(yields, this_t)
  }
}
yields %>% knitr::kable(digits=2)
```

```{r}
yields %>% 
  group_by(Gene, Group) %>% 
  summarise('Yield difference when gene present' = diff(`Mean yield`)) %>% 
  knitr::kable(digits=2)
```


# Manhattan plots

A reviewer asked for Manhattan plots. 

```{r}
# slightly modified from https://danielroelfs.com/blog/how-i-create-manhattan-plots-using-ggplot/
# Thanks :) 
myplot <- function(f) {
  print(f)
  mlm <- read_csv(paste('./output/GAPIT/', f, sep=''))
  mlm <- mlm %>% filter(SNP != 'GlymaLee.02G228600.1.p')
  data_cum <- mlm %>% 
    group_by(Chromosome) %>% 
    summarise(max_bp = max(Position)) %>% 
    mutate(bp_add = lag(cumsum(max_bp), default = 0)) %>% 
    select(Chromosome, bp_add)
  
  mlm <- mlm %>% 
    inner_join(data_cum, by = "Chromosome") %>% 
    mutate(bp_cum = Position + bp_add)
  
  axis_set <- mlm %>% 
    group_by(Chromosome) %>% 
    summarize(center = mean(bp_cum))
  
  ylim <- mlm %>% 
     filter(`FDR_Adjusted_P-values` == min(`FDR_Adjusted_P-values`)) %>% 
     mutate(ylim = abs(floor(log10(`FDR_Adjusted_P-values`))) + 1) %>% 
     pull(ylim)
  
  mlm_plot <- ggplot(mlm, aes(x = bp_cum, y = -log10(`FDR_Adjusted_P-values`), 
                color = as_factor(Chromosome))) +
    geom_hline(yintercept = -log10(0.05), color = "grey40", linetype = "dashed") + 
    geom_point(alpha = 0.75) +
    scale_x_continuous(label = axis_set$Chromosome, breaks = axis_set$center) +
    scale_y_continuous(expand = c(0,0), limits = c(0, ylim)) +
    scale_color_manual(values = rep(c("#276FBF", "#183059"), unique(length(axis_set$Chromosome)))) +
    labs(x = NULL, 
         y = "-log<sub>10</sub>(p)") + 
    theme_minimal() +
    theme( 
      legend.position = "none",
      panel.border = element_blank(),
      panel.grid.major.x = element_blank(),
      panel.grid.minor.x = element_blank(),
      axis.title.y = element_markdown(),
      #axis.text.x = element_text(angle = 60, size = 8, vjust = 0.5)
    )
mlm_plot
}
```

```{r}
files <- list.files(path='./output/GAPIT/', '*GWAS.Results.csv')
names(files) <- str_split(files, pattern = '\\.', simplify=T)[,2]
# remove MLM as there are no hits
files <- files[c('GLM','MLMM','FarmCPU')]
plot_list <- purrr::map(files, myplot)
```

```{r}
result <- plot_list$GLM /  plot_list$FarmCPU + plot_annotation(tag_levels = 'A')

result
```

```{r}
cowplot::save_plot(filename = 'output/GWAS_Manhattan.png',result, base_height = 5, base_width=8)
```

